### PR TITLE
[REVIEW] Add missing call to cudaStreamSynchronize() in get_value()

### PR DESCRIPTION
**Pull request by nvdbaranec**
Fixes  https://github.com/rapidsai/cudf/issues/6700

Asynchronous memcpy was returning a stack variable before synchronizing.

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/include/cudf/detail/get_value.cuh [hunk 1]
```diff
@@ -47,6 +47,7 @@ T get_value(column_view const& col_view, size_type element_index, cudaStream_t s
   T result;
   CUDA_TRY(cudaMemcpyAsync(
     &result, col_view.data<T>() + element_index, sizeof(T), cudaMemcpyDeviceToHost, stream));
+  CUDA_TRY(cudaStreamSynchronize(stream));
```
Comment (jrhemstad) [inline cpp/include/cudf/detail/get_value.cuh:50]: Can you add a note to the documentation that this function synchronizes the stream?

**Review by harrism — APPROVED** [review]

**Review by davidwendt — APPROVED** [review]

**Review by jrhemstad — APPROVED** [review]

File: cpp/include/cudf/detail/get_value.cuh [hunk 2]
```diff
@@ -26,7 +26,8 @@ namespace detail {
 /**
  * @brief Return a fixed-width value from column.
  *
- * This will retrieve the specified value from device memory.
+ * This will retrieve the specified value from device memory. This function
```
Comment (harrism) [inline cpp/include/cudf/detail/get_value.cuh:29]: One of my pet peeves is using future tense in documentation. Let's switch this to present tense, it is nearly always clearer. Plus, with your addition the tenses are mixed.

I know, I know, this is very nit-picky.

```suggestion
 * Retrieves the specified value from device memory. This function
```

**Review by harrism — CHANGES_REQUESTED** [review]

**Comment by nvdbaranec** [comment]
rerun tests

**Review by harrism — APPROVED** [review]

**Comment by nvdbaranec** [comment]
rerun tests

**Comment by nvdbaranec** [comment]
rerun tests

**Comment by nvdbaranec** [comment]
rerun tests

**Pull request merged**